plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/TestBloomFilterStore.java

```diff
@@ -0,0 +1,313 @@
+        }
+    }
+
+    private static BloomFilterStore generateBloomFilterStore(ParquetTester.TempFile tempFile, boolean enableBloomFilter, List<Object> testValues, ObjectInspector objectInspector)
+            throws Exception
+    {
+        List<ObjectInspector> objectInspectors = singletonList(objectInspector);
+        List<String> columnNames = ImmutableList.of(COLUMN_NAME);
+
+        JobConf jobConf = new JobConf(newEmptyConfiguration());
+        jobConf.setEnum(WRITER_VERSION, PARQUET_1_0);
+        jobConf.setBoolean(BLOOM_FILTER_ENABLED, enableBloomFilter);
+
+        ParquetTester.writeParquetColumn(
+                jobConf,
+                tempFile.getFile(),
+                CompressionCodecName.SNAPPY,
+                ParquetTester.createTableProperties(columnNames, objectInspectors),
+                getStandardStructObjectInspector(columnNames, objectInspectors),
+                new Iterator<?>[] {testValues.iterator()},
+                Optional.empty(),
+                false,
+                DateTimeZone.getDefault());
+
+        TrinoFileSystemFactory fileSystemFactory = new HdfsFileSystemFactory(HDFS_ENVIRONMENT);
+        TrinoFileSystem fileSystem = fileSystemFactory.create(getHiveSession(new HiveConfig().setHiveStorageFormat(HiveStorageFormat.PARQUET)));
+        TrinoInputFile inputFile = fileSystem.newInputFile(tempFile.getFile().getPath());
+        TrinoParquetDataSource dataSource = new TrinoParquetDataSource(inputFile, new ParquetReaderOptions(), new FileFormatDataSourceStats());
+
+        ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, Optional.empty());
+        ColumnChunkMetaData columnChunkMetaData = getOnlyElement(getOnlyElement(parquetMetadata.getBlocks()).getColumns());
+
+        return new BloomFilterStore(dataSource, getOnlyElement(parquetMetadata.getBlocks()), Set.of(columnChunkMetaData.getPath()));
+    }
+
+    private static class BloomFilterTypeTestCase
+    {
+        private final List<Object> matchingValues;
+        private final List<Object> nonMatchingValues;
+        private final List<Object> writeValues;
+        private final Type sqlType;
+        private final ObjectInspector objectInspector;
+
+        private BloomFilterTypeTestCase(List<Object> writeValues, List<Object> nonMatchingValues, Type sqlType, ObjectInspector objectInspector)
+        {
+            this.sqlType = requireNonNull(sqlType);
+            this.objectInspector = requireNonNull(objectInspector);
+            this.writeValues = requireNonNull(writeValues);
+
+            this.matchingValues = writeValues.stream()
+                    .map(data -> toNativeContainerValue(sqlType, data))
+                    .collect(toImmutableList());
+            this.nonMatchingValues = nonMatchingValues.stream()
+                    .map(data -> toNativeContainerValue(sqlType, data))
+                    .collect(toImmutableList());
+        }
```

Comment:
Please add
```
@Override
        public String toString()
        {
            return toStringHelper(this)
                    .add("writeValues", writeValues)
                    .add("sqlType", sqlType)
                    .toString();
        }
```

Reply:
I see, it would helps IntelliJ to show test cases, thanks for letting me know!
<img width="1181" alt="Screen Shot 2022-12-25 at 2 31 24 PM" src="https://user-images.githubusercontent.com/4726688/209459039-366a4bf1-58be-41a9-9104-eebfb59a413a.png">